fitbenchmarking/fitbenchmarking
# Use `eval_model` in bumps controller so that MaxRuntime timer is checked

#### Description of Work
This PR ensures the bumps controller uses the `eval_model` method instead of the `function` method so that the MaxRuntime timer gets checked will fitting is taking place. Previously this was not happening, and so bumps would never time out until the fitting calculation had finished (which could take a long time).

Having skimmed through the other controllers, I think this is a bumps-only issue.

Fixes #930

#### Testing Instructions

1. Follow the instructions in the issue. The max runtime timeout should happen in roughly the expected amount of time.

2. Its worth having a quick check of the other controllers to make sure `eval_model` is called during their fitting.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
